appeals/web/src/client/components/file-uploader/_client-actions.js

```diff
@@ -336,6 +407,22 @@ const clientActions = (uploadForm) => {
 		window.removeEventListener('beforeunload', leavePageWarningEventHandler);
 	};
 
+	/**
+	 * @param {string|undefined} caseReference
+	 * @param {string} fileGuid
+	 * @param {string} fileName
+	 * @param {string} [latestVersion]
+	 * @returns {string}
+	 */
+	const createBlobStorageUrl = (caseReference, fileGuid, fileName, latestVersion) => {
```

Comment:
I'm thinking this could be an issue with data integrity... Maybe we should be more drastic and save the file in storage with a GUID instead of v{version}: https://domain/container/appeal/reference/documentGuid/versionGuid/fileName.ext

Then let the database store the version number, against that unique documentURI...

Reply:
As discussed, given the number of changes already in this PR, I'd prefer to do this as a separate task